apps/framework-cli/src/cli/routines.rs

```diff
@@ -263,29 +263,34 @@ async fn initialize_project_state(
     let producer = redpanda::create_producer(project.redpanda_config.clone());
 
     info!("Starting schema directory crawl...");
-    let crawl_result =
-        process_schemas_in_dir(&schema_dir, project, &configured_client, route_table).await;
-
-    let _ = post_current_state_to_console(
-        project,
-        &configured_client,
-        &producer,
-        route_table.clone(),
-        project.console_config.clone(),
-    )
-    .await;
-
-    match crawl_result {
-        Ok(_) => {
-            info!("Schema directory crawl completed successfully");
-            Ok(())
-        }
-        Err(e) => {
-            debug!("Schema directory crawl failed");
-            debug!("Error: {:?}", e);
-            Err(e)
+
+    with_spinner_async("Processing schema file", async {
+        let crawl_result =
+            process_schemas_in_dir(&schema_dir, project, &configured_client, route_table).await;
+
+        let _ = post_current_state_to_console(
+            project,
+            &configured_client,
+            &producer,
+            route_table.clone(),
+            project.console_config.clone(),
+        )
+        .await;
+
+        match crawl_result {
+            Ok(_) => {
+                info!("Schema directory crawl completed successfully");
+                Ok(())
+            }
+            Err(e) => {
+                debug!("Schema directory crawl failed");
+                debug!("Error: {:?}", e);
+                Err(e)
+            }
         }
-    }
+    })
+    .await?;
+    Ok(())
```

Comment:
looks like we can simplify this and remove `; Ok(())`

Reply:
@phiSgr Removing the Ok(()) results in : `expected Result<(), Error>, found ()`
As it stands I can't simply remove the `Ok(())`